contracts/src/libraries/LPMath.sol

```diff
@@ -127,6 +127,124 @@ library LPMath {
+            FixedPointMath.max(_before, 0);
+        if (delta > 0) {
+            _longExposure += uint256(delta);
+        } else if (delta < 0) {
+            _longExposure -= uint256(-delta);
+        }
+
+        return _longExposure;
+    }
+
+    /// @dev Verifies that the price discovery is valid after liquidity is added
+    ///      by checking if the pool is solvent after opening the largest long
+    ///      possible on the curve and ensuring that the pool is still solvent.
+    /// @param _shareReserves
+    /// @param _shareAdjustment
+    /// @param _bondReserves
+    /// @param _minimumShareReserves
+    /// @param _initialVaultSharePrice
+    /// @param _vaultSharePrice
+    /// @param _timeStretch
+    /// @param _checkpointExposure
+    /// @param _longExposure
+    /// @param _shareReserves The share reserves.
+    /// @param _shareAdjustment The share adjustment.
+    /// @param _bondReserves The bond reserves.
+    /// @param _minimumShareReserves The minimum share reserves.
+    /// @param _initialVaultSharePrice The initial vault share price.
+    /// @param _vaultSharePrice The vault share price.
+    /// @param _timeStretch The time stretch.
+    /// @param _checkpointExposure The checkpoint exposure.
+    /// @param _longExposure The global long exposure.
+    /// @return A flag indicating if the price discovery is valid.
+    function verifyPriceDiscovery(
+        uint256 _shareReserves,
+        int256 _shareAdjustment,
+        uint256 _bondReserves,
+        uint256 _minimumShareReserves,
+        uint256 _initialVaultSharePrice,
+        uint256 _vaultSharePrice,
+        uint256 _timeStretch,
+        int256 _checkpointExposure,
+        uint256 _longExposure
+    ) external pure returns (bool) {
+        // Calculate the share payment and bond proceeds of opening the
+        // largest possible long on the YieldSpace curve. This does not
+        // include fees.
+        (uint256 effectiveShareReserves, bool success) = HyperdriveMath
+            .calculateEffectiveShareReservesSafe(
+                _shareReserves,
+                _shareAdjustment
+            );
+        if (!success) {
+            return false;
+        }
+        uint256 maxSharePayment;
+        (maxSharePayment, success) = YieldSpaceMath.calculateMaxBuySharesInSafe(
+            effectiveShareReserves,
+            _bondReserves,
+            ONE - _timeStretch,
+            _vaultSharePrice,
+            _initialVaultSharePrice
+        );
+
+        uint256 maxBondProceeds;
+        (maxBondProceeds, success) = YieldSpaceMath.calculateMaxBuyBondsOutSafe(
+            effectiveShareReserves,
+            _bondReserves,
+            ONE - _timeStretch,
+            _vaultSharePrice,
+            _initialVaultSharePrice
+        );
+
+        // If one fails they should both be zero.
+        if (maxSharePayment == 0 || maxBondProceeds == 0) {
+            maxSharePayment = 0;
+            maxBondProceeds = 0;
```

Comment:
I think calculations should follow the logic from `_calculateOpenLong` where bond reserve delta is given derived from the given share reserve delta.